### PR TITLE
Feature/department id casting

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
## PR Overview
**This PR will address the following Issue/Feature:** This is a complimentary PR to the dbt_quickbooks [PR#82](https://github.com/fivetran/dbt_quickbooks/pull/82) that adds the department_id to the GL model.

**This PR will result in the following new package version:** `v0.8.0` 

Although this is not a breaking change, it will be batched with other breaking changes.

**Please detail what change(s) this PR introduces and any additional information that should be known during the review of this PR:**

This PR includes string type casting for the `department_id` in the relevant header and line item tables. 
  - `stg_quickbooks__bill`
  - `stg_quickbooks__bill_payment`
  - `stg_quickbooks__credit_memo`
  - `stg_quickbooks__department`
  - `stg_quickbooks__deposit`
  - `stg_quickbooks__invoice`
  - `stg_quickbooks__journal_entry_line`
  - `stg_quickbooks__purchase`
  - `stg_quickbooks__refund_receipt`
  - `stg_quickbooks__sales_receipt`
  - `stg_quickbooks__vendor_credit`

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt compile
- [x] dbt run –full-refresh
- [x] dbt run
- [x] dbt test
- [x] dbt run –vars (if applicable)

Before marking this PR as “ready for review” the following have been applied:
- [x] The appropriate issue has been linked and tagged
- [x] You are assigned to the corresponding issue and this PR
- [x] BuildKite integration tests are passing

### Detailed Validation
Please acknowledge that the following validation checks have been performed prior to marking this PR as “ready for review”:
- [x] You have validated these changes and assure this PR will address the respective Issue/Feature.
- [x] You are reasonably confident these changes will not impact any other components of this package or any dependent packages.
- [x] You have provided details below around the validation steps performed to gain confidence in these changes.

This is a pretty straight forward update where the `department_id` fields are casted in the relevant staging models in order to ensure the unioning of the fields downstream are successful. This casting is particularly necessary for tables that have all `null` values as the datatype cannot always be inferred. Therefore, the casting is necessary to ensure success.

Additionally, it was confirmed that all relevant macros have the `department_id` and set to datatype string. Therefore we can be comfortable that the field will be included regardless.

### Standard Updates
Please acknowledge that your PR contains the following standard updates:
- Package versioning has been appropriately indexed in the following locations:
    - [x] indexed within dbt_project.yml
    - [x] indexed within integration_tests/dbt_project.yml
- [x] CHANGELOG has individual entries for each respective change in this PR
    
- [x] README updates have been applied (if applicable)
    
- [x] DECISIONLOG updates have been updated (if applicable)
- [x] Appropriate yml documentation has been added (if applicable)

### dbt Docs
Please acknowledge that after the above were all completed the below were applied to your branch:
- [x] docs were regenerated (unless this PR does not include any code or yml updates)

### If you had to summarize this PR in an emoji, which would it be?
 
🧵

### CODE REVIEW

**Review by fivetran-avinash — CHANGES_REQUESTED** [review]
@fivetran-joemarkiewicz I noticed that the seed files had mostly null department ids so I tried and tested with and without a department id to make sure everything looked good (passed both on the source and transform so all looks good there).

Only one significant comment to address and you should be good to go!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,19 @@
+# dbt_quickbooks_source v0.8.0
+[PR #44](https://github.com/fivetran/dbt_quickbooks_source/pull/44) includes the following updates:
+## Under the Hood
+- Casted the `department_id` within the following staging models to be of type string. This ensures downstream unions on this type are consistent and of the same type.
+  - `stg_quickbooks__bill`
+  - `stg_quickbooks__bill_payment`
+  - `stg_quickbooks__credit_memo`
+  - `stg_quickbooks__department`
+  - `stg_quickbooks__deposit`
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply CHANGELOG.md:9]: No reason other than it is not brought downstream into the GL model so there wasn't a critical need to cast accordingly.

I think we should cast it to remain consistent. I just updated the model and changelog accordingly. Thanks for calling this out!

**Comment by fivetran-joemarkiewicz** [comment]
Thanks for the review @fivetran-avinash! Applied the changes and all is ready for re-review!

**Review by fivetran-avinash — APPROVED** [review]
Time to merge!

**Pull request merged**